File: benchmark-java-node-php-python/php-laravel-octane/monolith/app/Services/BonusService.php
<?php

namespace App\Services;

use App\Models\Bonus;
use App\Models\Client;
use Carbon\Carbon;
use Illuminate\Http\Request;

class BonusService
{
    public function createBonus(array $data)
    {
        // 1. Integrity Check
        $client = Client::find($data['clientId']);

        if (!$client) {
            return ['error' => 'Client not found', 'status' => 404];
        }

        if (!$client->active) {
            return ['error' => 'Client is inactive', 'status' => 400];
        }

        // 2. Biz Logic
        $amount = $data['amount'];
        if ($amount > 100) {
            $amount = $amount * 1.1;
        }

        $now = Carbon::now();
        $expirationDate = $now->copy()->addDays(30);

        $bonus = Bonus::create([
            'amount' => $amount,
            'description' => 'PHPLARAVELOCTANE - ' . $data['description'],
            'client_id' => $data['clientId'],
            'expiration_date' => $expirationDate,
            'created_at' => $now
        ]);

        return ['data' => $bonus, 'status' => 201];
    }

    public function getBonus($id)
    {
        $bonus = Bonus::find($id);
        if (!$bonus) {
            return ['error' => 'Not found', 'status' => 404];
        }
        return ['data' => $bonus, 'status' => 200];
    }
}

File: benchmark-java-node-php-python/cpp-drogon/monolith/src/services/BonusService.cpp
#include "../../include/services/BonusService.h"
#include "../../include/repositories/BonusRepository.h"
#include "../../include/repositories/ClientRepository.h"
#include <chrono>
#include <stdexcept>

namespace services {

std::shared_ptr<models::Bonus> BonusService::createBonus(const dto::BonusDTO& dto) {
    // Validate input
    if (!dto.isValid()) {
        throw std::invalid_argument(dto.getValidationError());
    }
    
    // Check if client exists and is active
    auto client = clientRepository_.findById(dto.clientId);
    if (!client) {
        throw std::runtime_error("Client not found");
    }
    
    if (!client->getActive()) {
        throw std::runtime_error("Client is inactive");
    }
    
    // Calculate final amount with business logic
    double finalAmount = calculateFinalAmount(dto.amount);
    
    // Create bonus entity
    models::Bonus bonus;
    bonus.setAmount(finalAmount);
    bonus.setDescription("CPPDROGON - " + dto.description);
    bonus.setClientId(dto.clientId);
    bonus.setCreatedAt(std::chrono::system_clock::now());
    bonus.setExpirationDate(std::chrono::system_clock::now() + std::chrono::hours(30 * 24)); // 30 days
    
    return bonusRepository_.create(bonus);
}

std::shared_ptr<models::Bonus> BonusService::getBonus(int64_t id) {
    return bonusRepository_.findById(id);
}

double BonusService::calculateFinalAmount(double amount) const {
    if (amount > 100.0) {
        return amount * 1.1; // 10% bonus for amounts > 100
    }
    return amount;
}

} // namespace services
